### PR TITLE
Fix everything lgtm was complaining about

### PR DESCRIPTION
**Pull request by chunkybanana**

### CODE REVIEW

**Comment by cgccuser** [comment]
Looks like you forgot to import math somewhere?

File: vyxal/helpers.py [hunk 1]
```diff
@@ -8,7 +8,6 @@
 import collections
 import inspect
 import itertools
-import math
```
Comment (cgccuser) [inline vyxal/helpers.py:11]: why?

**Review by cgccuser — CHANGES_REQUESTED** [review]

**Review by Lyxal — CHANGES_REQUESTED** [review]
The test cases need that math import. Leave it in helpers because I think there's a few other things that need it (mostly Sympy to python stuff)

**Comment by pxeger** [comment]
I can't see anywhere in the tests themselves that use `math`; it seems to be just the sympy conversion things. Besides, it's bad practice to change your main code to make tests easier

**Comment by Lyxal** [comment]
> I can't see anywhere in the tests themselves that use `math`; it seems to be just the sympy conversion things. Besides, it's bad practice to change your main code to make tests easier

That math import is also needed for the main files too.

File: vyxal/helpers.py (hunk 1 above)
Comment (chunkybanana) [reply vyxal/helpers.py:11]: Lgtm said it was an unnecessary import, and a quick ctrl-f in there found nothing but description comments.

**Comment by chunkybanana** [comment]
> > I can't see anywhere in the tests themselves that use `math`; it seems to be just the sympy conversion things. Besides, it's bad practice to change your main code to make tests easier
> 
> That math import is also needed for the main files too - it's used for converting Sympy stuff to decimals using python precision

My point is, nothing in `helpers.py` directly uses `math`, so it doesn't need to be there.

**Comment by Lyxal** [comment]
Except it can't.

```
exec(sympy.pycode(expr))
```

Will sometimes contain references to the math library. It does not contain an import statement however.

This is so that Floats can be displayed with things like that special flag without having an unruly amount of 0s at the end, and also means that complex numbers are converted to their python equivalent for display and for testing purposes.

**Comment by Lyxal** [comment]
This is one of those cases where it's best to just slip one of those `ignore issue` comments into the source.

**Comment by cgccuser** [comment]
Can we not `import math` right above `exec(sympy.pycode(expr))`?

**Comment by cgccuser** [comment]
Or can we use `value.evalf()` instead of `eval(sympy.py_code(value))`?

**Comment by chunkybanana** [comment]
> Except it can't.
> 
> ```
> exec(sympy.pycode(expr))
> ```
> 
> Will sometimes contain references to the math library. It does not contain an import statement however.
> 
> This is so that Floats can be displayed with things like that special flag without having an unruly amount of 0s at the end, and also means that complex numbers are converted to their python equivalent for display and for testing purposes.

Oh ok.

**Comment by Lyxal** [comment]
> # lgtm [py/unused-import]

not without potentially having a lot of trailing 0s for numbers like `0.5`

**Review by Lyxal — APPROVED** [review]
very happy now

**Review by cgccuser — APPROVED** [review]
Oh well, I'm not happy about evaling strings by sympy but if it's the best option we have, let's go ahead with this.

**Pull request merged**